runtime/storage/cache/cache.go

```diff
@@ -0,0 +1,267 @@
+// Package cache provides the ability to define distributed Redis cache clusters
+// and functionality to use them in a fully type-safe manner.
+//
+// For more information see https://encore.dev/docs/develop/caching
+package cache
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/go-redis/redis/v8"
+)
+
+// ClusterConfig represents the configuration of cache clusters.
+type ClusterConfig struct {
+	// EvictionPolicy decides how the cache evicts existing keys
+	// to make room for new data when the max memory limit is reached.
+	//
+	// If not specified the cache defaults to AllKeysLRU.
+	EvictionPolicy EvictionPolicy
+}
+
+// An EvictionPolicy describes how the cache evicts keys to make room for new data
+// when the maximum memory limit is reached.
+//
+// See https://redis.io/docs/manual/eviction/#eviction-policies for more information.
+type EvictionPolicy string
+
+// NOTE: These values need to be added to the runtimeconstants package
+// and the parser package for the parser to be aware of them.
+
+const (
```

Comment:
I'm not sure I understand fully the difference between `AllKeysX` and `VolatileX`. I've read the doc's and I think by setting Volatile it means that only keys with an expiration will be deleted and keys without an expiration will never be deleted.

> if there are no keys to evict with an expiration set.
This bit then confuses me, I think the wording would be better:

> If no keys within the cache have an expiration set, then this setting behaves like `NoEviction`.

Reply:
That's correct. We can try to improve the docs